Tasks/ServiceFabricComposeDeploy/ServiceFabricComposeDeploy.ps1

```diff
@@ -163,48 +163,101 @@ try {
         $getStatusParameters['TimeoutSec'] = $getStatusTimeoutSec
     }
 
+    $upgrading = $false
     $existingApplication = Get-ServiceFabricComposeApplicationStatusHelper -ApiVersion $apiVersion -GetStatusParameters $getStatusParameters
     if ($existingApplication -ne $null)
     {
-        Write-Host (Get-VstsLocString -Key RemovingApplication -ArgumentList $applicationName)
-
-        Remove-ServiceFabricComposeApplicationHelper -ApiVersion $apiVersion -RemoveParameters $removeParameters
-        do
+        if ($apiVersion -eq "2.8")
         {
-            Write-Host (Get-VstsLocString -Key CurrentStatus -ArgumentList $existingApplication.Status)
-            Start-Sleep -Seconds 3
-            $existingApplication = Get-ServiceFabricComposeApplicationStatusHelper -ApiVersion $apiVersion -GetStatusParameters $getStatusParameters
+            $upgrading = $true
         }
-        while ($existingApplication -ne $null)
-        Write-Host (Get-VstsLocString -Key ApplicationRemoved)
-    }
+        else
+        {
+            Write-Host (Get-VstsLocString -Key RemovingApplication -ArgumentList $applicationName)
 
-    Write-Host (Get-VstsLocString -Key CreatingApplication)
-    New-ServiceFabricComposeApplicationHelper -ApiVersion $apiVersion -DeployParameters $deployParameters
+            Remove-ServiceFabricComposeApplicationHelper -ApiVersion $apiVersion -RemoveParameters $removeParameters
+            do
+            {
+                Write-Host (Get-VstsLocString -Key CurrentStatus -ArgumentList $existingApplication.Status)
+                Start-Sleep -Seconds 3
+                $existingApplication = Get-ServiceFabricComposeApplicationStatusHelper -ApiVersion $apiVersion -GetStatusParameters $getStatusParameters
+            }
+            while ($existingApplication -ne $null)
+            Write-Host (Get-VstsLocString -Key ApplicationRemoved)
+        }
+    }
 
-    Write-Host (Get-VstsLocString -Key WaitingForDeploy)
-    $newApplication = Get-ServiceFabricComposeApplicationStatusHelper -ApiVersion $apiVersion -GetStatusParameters $getStatusParameters
-    while (($newApplication -eq $null) -or `
-           ($newApplication.Status -eq 'Provisioning') -or `
-           ($newApplication.Status -eq 'Creating'))
+    if ($upgrading)
     {
-        if ($newApplication -eq $null)
+        $upgradeParameters = New-Object 'System.Collections.Hashtable' $deployParameters
+        $upgradeParameters.Monitored = $True
```

Comment:
Don't we need to take knobs as task inputs- just like in ServiceFabricDeploy task?

Reply:
Long term we may need to. The feature is still in preview on the platform side so we are getting the golden path working and waiting for feedback on what knobs user's want/need.